src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java

```diff
@@ -1630,4 +1686,17 @@ private void logAttackMoves(final List<ProTerritory> prioritizedTerritories) {
       }
     }
   }
+
+  private boolean canAirSafelyLandAfterAttack(final Unit unit, final Territory t) {
+    final boolean isAdjacentToAlliedFactory =
+        Matches.territoryHasNeighborMatching(data,
+            ProMatches.territoryHasInfraFactoryAndIsAlliedLand(player, data)).match(t);
+    final int range = TripleAUnit.get(unit).getMovementLeft();
+    final int distance =
+        data.getMap().getDistance_IgnoreEndForCondition(ProData.unitTerritoryMap.get(unit), t,
```

Comment:
I'd prefer to keep AI PRs focused just on AI code and not touch engine code. That should be a separate PR. In this case that boolean is passed much further down the chain. It will also make this method name even longer. I think its better to leave it for now.